### PR TITLE
Enable nested beforeTemplateIsBaked calls

### PR DESCRIPTION
**Pull request by codetheweb**
This is kinda crazy but necessary for Seam Connect work with the isolated worker database.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,57 @@ This works across the entire test suite.
...
+    t,
+    barDatabase.beforeTemplateIsBakedResult.fooTemplateName
+  )
+  await t.notThrowsAsync(async () => {
+    await fooDatabase.pool.query(`SELECT * FROM "foo"`)
+  })
+})
+```
+
+Although it's not shown in the above example, because of `ava-postgres`'s automatic de-duping by parameter combinations, any returned template name is "linked" to the parameters passed to the `getTestDatabase()` function.
```
Comment (codetheweb) [inline README.md:200]: wondering if there's a better way to phrase this

Comment (mxsdev) [reply README.md:200]: I would include a comment string with the actual DB name above the relevant parts to demonstrate this directly

**Review by mxsdev — APPROVED** [review]
🙏

**Review by panzerchris — APPROVED** [review]
Looking forward to learning more about this tomorrow!

File: README.md (hunk 1 above)
Comment (codetheweb) [reply README.md]: decided to just remove it for now

**Pull request merged**